jenkins-jobs/jobs/DesktopCommonJob.groovy

```diff
@@ -0,0 +1,96 @@
+def basePath = 'desktop-common'
+folder(basePath)
+
+def prJob = job("$basePath/desktop-common - PR") {
+    scm {
+        git {
+            remote {
+                url('https://github.com/wpilibsuite/desktop-common.git')
+                refspec('+refs/pull/*:refs/remotes/origin/pr/*')
+            }
+            // This is purposefully not a GString. This is a jenkins environment
+            // variable, not a groovy variable
+            branch('${sha1}')
+        }
+    }
+    triggers {
+        githubPullRequest {
+            admins(['333fred', 'PeterJohnson', 'bradamiller', 'Kevin-OConnor'])
+            orgWhitelist('wpilibsuite')
+            useGitHubHooks()
+        }
+    }
+}
+
+setupProperties(prJob)
+setupBuildSteps(prJob, false)
+
+def developmentJob = job("$basePath/desktop-common - Development") {
+    triggers {
+        scm('H/15 * * * *')
+    }
+    publishers {
+        downstream('Eclipse Plugins/Eclipse Plugins - Development')
```

Comment:
Nothing depends on this?

Reply:
Shuffleboard, OutlineViewer, and Pathweaver will all depend on it. They just can't yet because there's no publishing for desktop-common yet

Reply:
Although if there's a better way to rebuild the downstream apps, I'd be happy to use it